### PR TITLE
[MPAS-Ocean standalone] Allow land-ice topo to come from MPAS variables 

### PR DESCRIPTION
**Pull request by xylar**
This applies only to the `global_ocean` test case in `init` mode for standalone MPAS-Ocean.  E3SM simulations are unaffected.

In the same way that we could previously bypass the interpolation from a lat-lon file for bathymetry, this merge allows us to bypass the interpolation of variables related to land-ice topography and instead read them in from a stream.

[BFB]

### CODE REVIEW

File: components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F [hunk 1]
```diff
@@ -3039,23 +3048,26 @@ subroutine ocn_init_validate_global_ocean(configPool, packagePool, iocontext, iE
       end if
 
       if (config_global_ocean_depress_by_land_ice) then
-         if (trim(config_global_ocean_land_ice_topo_file) == 'none') then
+         if (config_global_ocean_land_ice_topo_file == 'none' .and. &
+             config_global_ocean_topography_source == 'latlon_file') then
             call mpas_log_write( 'Validation failed for global ocean. '// &
                'Invalid filename for config_global_ocean_land_ice_topo_file', MPAS_LOG_CRIT)
             iErr = 1
             return
```
Comment (xylar) [inline components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F:3056]: Previously, you had to provide a placeholder name for `config_global_ocean_land_ice_topo_file` even if you used `config_global_ocean_topography_source = 'mpas_variable'`.  With this change, you should no longer get an error when you don't provide a placeholder filename.

**Comment by xylar** [comment]
## Testing

I have run this in conjunction with https://github.com/MPAS-Dev/compass/pull/566.  The topography (bathymetry, ice draft, ice thickness and various masks) are remapped outside of MPAS-Ocean init mode using a conservative mapping file.  Then, the results are read in using `config_global_ocean_topography_source = 'mpas_variable'` from a stream.

**Comment by xylar** [comment]
@sbrus89 and @mark-petersen, I'd appreciate your review on this when you can.  It's somewhat urgent because we have a lot of steps to get working before we can make E3SM v3 meshes, and this is early in the chain.

**Comment by xylar** [comment]
@mark-petersen and @sbrus89, This now passes the compass `pr` suite so it's ready for review.

**Comment by xylar** [comment]
Thanks @mark-petersen!

**Review by sbrus89 — COMMENTED** [review]
@xylar, this is BFB for the tides test case. I just have one minor comment.

File: components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F [hunk 2]
```diff
@@ -815,40 +819,40 @@ subroutine ocn_init_setup_global_ocean_create_model_topo(domain, iErr)!{{{
...
               else
                  call mpas_log_write( 'Invalid choice of config_global_ocean_topography_method.', MPAS_LOG_CRIT)
                  iErr = 1
                  call mpas_dmpar_finalize(domain % dminfo)
               endif
-          else 
+          else
               ! Get bottomDepthObserved from bed_elevation variable
               bottomDepthObserved = bed_elevation
           endif
```
[the remark this answers is not in the document]
Comment (xylar) [reply components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F:858]: Yes, I think that's a good idea.  With an error message if there's an unexpected value.

[the remark this answers is not in the document]
Comment (xylar) [reply components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F:858]: The error message should go into init, so I'll do that and then just have the `else if` here.

[the remark this answers is not in the document]
Comment (xylar) [reply components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F]: I have made this change.

**Comment by xylar** [comment]
@sbrus89, could you have another look as soon as you're able?

**Review by sbrus89 — APPROVED** [review]
@xylar, this looks good to me. Thanks for adding the additional logic. I was also going to suggest the unexpected value error, so I like that change as well.

**Comment by jonbob** [comment]
passes sanity testing, merged to next

**Pull request merged**